### PR TITLE
Sm/record get where boolean w 6236756

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
turns strings that appear to be booleans into booleans 
tests with a NUT

### What issues does this PR fix or reference?
@W-6236756@

### CODE REVIEW

**Comment by WillieRuemmele** [comment]
has this always been a bug? I saw the chatter thread was 2 years old... just surprised no one else has noticed this

**Review by WillieRuemmele — APPROVED** [review]

**Comment by mshanemc** [comment]
> has this always been a bug? I saw the chatter thread was 2 years old... just surprised no one else has noticed this

yes.  I just don't think there's that many people using the command.

File: packages/plugin-data/src/dataCommand.ts [hunk 1]
```diff
@@ -169,13 +169,19 @@ export abstract class DataCommand extends SfdxCommand {
...
-        constructedObject[key] = pair.substr(eqPosition + 1);
+        constructedObject[key] = this.convertToBooleanIfApplicable(pair.substr(eqPosition + 1));
       }
     });
 
     return constructedObject;
   }
 
+  private convertToBooleanIfApplicable(input: string): boolean | string {
+    if (input.toLowerCase() === 'false') return false;
```
Comment (peternhale) [inline packages/plugin-data/src/dataCommand.ts:180]: @mshanemc You may want to trim the input. Not sure how strict key/value pairs are treated on the command line. 
```
sfdx force:data:record:get -s Account -w "Name='Universal Containers' Foo=true"
sfdx force:data:record:get -s Account -w "Name='Universal Containers' Foo= true"
```

**Review by peternhale — CHANGES_REQUESTED** [review]

File: packages/plugin-data/src/dataCommand.ts (hunk 1 above)
Comment (mshanemc) [reply packages/plugin-data/src/dataCommand.ts:180]: good point.  I put it in `transformKeyValueSequence`

Comment (peternhale) [reply packages/plugin-data/src/dataCommand.ts:180]: @mshanemc Not sure that is the right place. What if there are intentional leading or trailing whitespace is string values?

**Review by peternhale — CHANGES_REQUESTED** [review]

**Review by peternhale — APPROVED** [review]

**Pull request merged**